app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/GetNinoController.scala

```diff
@@ -0,0 +1,98 @@
+import uk.gov.hmrc.eoricommoncomponent.frontend.controllers.auth.AuthAction
+import uk.gov.hmrc.eoricommoncomponent.frontend.controllers.registration.routes.ConfirmContactDetailsController
+import uk.gov.hmrc.eoricommoncomponent.frontend.domain._
+import uk.gov.hmrc.eoricommoncomponent.frontend.domain.messaging.Individual
+import uk.gov.hmrc.eoricommoncomponent.frontend.forms.MatchingForms.subscriptionNinoForm
+import uk.gov.hmrc.eoricommoncomponent.frontend.models.{Journey, Service}
+import uk.gov.hmrc.eoricommoncomponent.frontend.services.registration.MatchingService
+import uk.gov.hmrc.eoricommoncomponent.frontend.services.subscription.SubscriptionDetailsService
+import uk.gov.hmrc.eoricommoncomponent.frontend.views.html.migration.how_can_we_identify_you_nino
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class GetNinoController @Inject() (
+  authAction: AuthAction,
+  matchingService: MatchingService,
+  mcc: MessagesControllerComponents,
+  matchNinoRowIndividualView: how_can_we_identify_you_nino,
+  subscriptionDetailsService: SubscriptionDetailsService
+)(implicit ec: ExecutionContext)
+    extends CdsController(mcc) {
+
+  def displayForm(service: Service, journey: Journey.Value): Action[AnyContent] =
+    authAction.ggAuthorisedUserWithEnrolmentsAction { implicit request => _: LoggedInUserWithEnrolments =>
+      Future.successful(
+        Ok(matchNinoRowIndividualView(subscriptionNinoForm, false, routes.GetNinoController.submit(service, journey)))
+      )
+    }
+
+  def submit(service: Service, journey: Journey.Value): Action[AnyContent] =
+    authAction.ggAuthorisedUserWithEnrolmentsAction {
+      implicit request => loggedInUser: LoggedInUserWithEnrolments =>
+        subscriptionNinoForm.bindFromRequest.fold(
+          formWithErrors =>
+            Future.successful(
+              BadRequest(
+                matchNinoRowIndividualView(formWithErrors, false, routes.GetNinoController.submit(service, journey))
+              )
+            ),
+          formData =>
+            matchIndividual(Nino(formData.id), service, journey, formData, InternalId(loggedInUser.internalId))
+        )
+    }
+
+  private def matchIndividual(
+    id: CustomsId,
+    service: Service,
+    journey: Journey.Value,
+    formData: IdMatchModel,
+    internalId: InternalId
+  )(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Result] =
+    subscriptionDetailsService.cachedNameDobDetails flatMap {
+      case Some(details) =>
+        matchingService
+          .matchIndividualWithId(
+            id,
+            Individual.withLocalDate(details.firstName, details.middleName, details.lastName, details.dateOfBirth),
+            internalId
+          )
+          .map {
+            case true  => Redirect(ConfirmContactDetailsController.form(service, journey))
+            case false => matchNotFoundBadRequest(formData, service, journey)
+          }
```

Comment:
For Booleans is better to use `if` than `match`.

`If` is faster and more readable for Booleans

Reply:
Copied from the "original" controller.  But I'll change